### PR TITLE
Finish Record Endpoints

### PR DESCRIPTION
**Pull request by lao9**
hey @josh-works , here's a breakdown of how the endpoints work here (all endpoint spec harness tests pass!): 

1. Basic Index and Show Controllers

very simple: there's an index and show - these point to the jbuilder views

**[note: I had to update the schema and a few of our own tests to make sure the basic spec harness tests pass for transactions and customers, so be sure to drop/re-migrate your database before running tests again]**

2. Find? 

So this takes a key value and just uses a basic "find_by" - since it only pulls out a single instance, we use the "show" in a new fancy controller called "FindController"

So first there's a before_action to properly format the params because the spec harness wants to get dates like this: "2012-03-27T14:54:03.000Z" and unit price like this: "174.00"

So referring to the application controller, we first determine if we need to format the params by checking if it's a created_at, updated_at, or unit_price param in the "format_params" method, which will then refer it to the next appropriate method. 

Then in that next method (if deemed necessary), we use a `params.merge!` which will permanently override the params the user provided and put the new one we want (which will change dates to DateTime objects and prices to an integer in pennies) we so can properly query using ActiveRecord. 

Then once the params are reformatted, we use the private method "search_params" and simply pass that into our find_by command to return the appropriate instance to pass to our jbuilder view.

3. Find_All?

This is EXACTLY THE SAME as Find? except we use ActiveRecord "where" in the index method of our FindController, which returns an array (even if it only returns one instance) of instances. Same process where we format the user query before passing it into our index where method, and then our index jbuilder view.

4. Random

Here there's still room to refactor this into a single method that can be used by all models, but right now each model has a "self.random" method. 

We have another special "RandomController" that just has a show method. Here we just run the random method to pluck a random instance of our entity and pass that into a show json jbuilder view.

Let me know if you have any other questions about this!

### CODE REVIEW

File: app/controllers/api/v1/customers/customers_random_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Api::V1::Customers::CustomersRandomController < ApplicationController
+
+  def show
+    @customer = Customer.random
```
Comment (case-eee) [inline app/controllers/api/v1/customers/customers_random_controller.rb:4]: nice 🎉

File: app/models/transaction.rb [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
-
   validates :invoice, presence: true
   validates :credit_card_number, presence: true
   validates :result, presence: true
 
   belongs_to :invoice
 
-  enum status: [failed: 0, success: 1]
-
+  def self.random
```
Comment (case-eee) [inline app/models/transaction.rb:8]: will all of your models have this method? if so, could we put it in `ApplicationRecord` to help DRY our models up?

File: spec/requests/api/v1/customers/customers_find_all_spec.rb [hunk 3]
```diff
@@ -0,0 +1,77 @@
+require 'rails_helper'
+
+describe "find many customers with params" do
```
Comment (case-eee) [inline spec/requests/api/v1/customers/customers_find_all_spec.rb:3]: tests are looking good 👍

**Review by case-eee — COMMENTED** [review]
@lao9 @josh-works great work! left you a few comments 👍

**Comment by case-eee** [comment]
oh also @lao9 - be sure you're testing each `random` method you've defined at the model level 👍

**Comment by josh-works** [comment]
Added a waffle card for testing `#random` at model level.

**Pull request merged**